cli/cli_kafka_runner_test.go

```diff
@@ -0,0 +1,343 @@
+			subpath = kafka.Avro
+		default:
+			t.Fatalf("unsupported encoding type")
+		}
+		KafkaConfig := confPathPrefix + subpath + "/" + test.ConfigFile
+		DataFile := dataPathPrefix + subpath + "/" + test.DataFile
+
+		// create command and kafka runner
+		fbsql := NewCommand(logger.StderrLogger)
+		setupKafkaRunnerCommand(t, fbsql)
+
+		// read config file so we can build a table before inserting data
+		config, err := kafka.ConfigStructFromFile(KafkaConfig)
+		if err != nil {
+			t.Fatalf("building kafka.Config struct: %s", err)
+		}
+
+		// create table to write to (kafka runner does not currently create tables)
+		createTable := strings.NewReader(fmt.Sprintf("CREATE TABLE %s %s;", config.Table, test.CreateTableStmt))
+		_, err = fbsql.Queryer.Query(fbsql.organizationID, fbsql.databaseID, createTable)
+		if err != nil {
+			t.Fatalf("creating table: %s", err)
+		}
+
+		// delete all tables after test (note this is after all tests not after each test)
+		defer func() {
+			dropTable := strings.NewReader(fmt.Sprintf("DROP TABLE %s;", config.Table))
+			_, err = fbsql.Queryer.Query(fbsql.organizationID, fbsql.databaseID, dropTable)
+			if err != nil {
+				t.Errorf("dropping table: %s", err)
+			}
+		}()
+
+		// create runner and hijack hosts based on test env (i.e. not from configuration file or other)
+		kafkaRunner, err := fbsql.newKafkaRunner(KafkaConfig)
+		if err != nil {
+			t.Fatalf("creating runner: %s", err)
+		}
+		kafkaRunner.FeaturebaseHosts = []string{pilosaHost}
+		kafkaRunner.SchemaRegistryURL = registryHost
+		kafkaRunner.KafkaBootstrapServers = []string{kafkaHost}
+		kafkaRunner.FeaturebaseGRPCHosts = []string{pilosaGrpcHost}
+
+		// read data that will go to kafka
+		var records []map[string]interface{}
+		records, err = recordsFromFile(DataFile)
+		if err != nil {
+			t.Errorf("reading raw messages from data file: %s", err)
+		}
+
+		// write data to kafka
+		var messages [][]byte
+		switch encode := test.Encode; encode {
+		case kafka.JSON:
+			messages, err = encodeJSONMessages(records)
+		case kafka.Avro:
+			t.Errorf("unsupported encoding type")
+		default:
+			t.Errorf("unsupported encoding type")
+		}
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		err = internal.ProduceMessages(kafkaRunner.Topics[0], strings.Join(kafkaRunner.KafkaBootstrapServers, ","), messages)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer func() {
+			// and delete the topic
+			err = internal.DeleteTopic(kafkaRunner.Topics[0], strings.Join(kafkaRunner.KafkaBootstrapServers, ","))
+			if err != nil {
+				t.Fatal(err)
+			}
+		}()
+
+		// Run it!!
+		kafkaRunner.MaxMsgs = uint64(len(messages))
+		err = kafkaRunner.Run()
+		if err != nil {
+			t.Fatalf("running runner: %s", err)
+		}
+
+		// and check the results...
+		for _, q := range test.Tests {
+
+			client := &http.Client{}
+			var data = strings.NewReader(q.Query)
+			req, err := http.NewRequest("POST", fmt.Sprintf("http://localhost:10101/index/%s/query", kafkaRunner.Index), data)
```

Comment:
This url to featurebase classic query endpoint seems problematic. I'm assuming these tests don't work in serverless. It seems like the query should just fbsql to run sql instead of pql commands.

Reply:
I made the PQL decision quickly after having issues comparing SQL results but have this working now with  the /sql endpoint.